anycompress/README.md

```diff
@@ -0,0 +1,19 @@
+## anycompress
+
+A transparent layer over [tendermint/tm-db.DB]() that wraps a database and intercepts .Set and .Get and compresses and decompresses google.protobuf.Any to reduce on the amount of redundant stored, regardless of the amount of compression by databases. The benefits of using anycompress come from the fac that database compression rarely happens in blocks after rows are stored, because post storage compressions reduce random accesses and don't fit well with the underlying data structures.
+
+### Examining benchmarks and savings
+```shell
+$ go run -tags rocksdb,boltdb,leveldb cmd/anycompress/main.go -n 1000 -type rocksdb
+
+time to generate  800000 key-value pairs: 4.009872537s
+Wrote MemoryProfile to disk mem-1598472491
+Wrote CPUProfile to disk cpu-1598472466
+Wrote MemoryProfile to disk mem-1598472536
+Wrote CPUProfile to disk cpu-1598472516
+Wrote MemoryProfile to disk mem-1598472581
+rocksdb savings: 53.068%
+Original:      736.308MiB (772075015B) - 2m15.992923978s
+AnyCompressed: 345.565MiB (362351474B) - 2m26.682974595s
+TimeSpent: 2m26.683016742s
+```
```

Comment:
Thanks for stats. I'm reviewing this code.

Could you please provide more background about this numbers:
+ it looks that rocksdb saving is exactly what the AnyCompress is giving.
+ did you test any DB compression level?
+ what's the finding with using gzip (it's used in the code).

Thanks.

Reply:
>it looks that rocksdb saving is exactly what the AnyCompress is giving.

Perhaps am mistaken by what you mean but "rocksdb saving" means "with rocksdb as a backend, adding AnyCompress saves 53.068% than if AnyCompress wasn't used"

> did you test any DB compression level?

I used the defaults as they are in tm-db, which is what the common user will use. I don't think tm-db allows that kind of tweaking? Would the purpose be here to find the optimal changes in tm-db for Rocksdb and tweak that accordingly?

>what's the finding with using gzip (it's used in the code).

None, gzip is only used in the benchmarking code to retrieve protobuf descriptions since the proto compiler serializes raw gzipped compressed bytes in generated *.pb.go files.

Thank you for the patience and for the questions @robert-zaremba!